pkg/mounter/safe_mounter_windows.go

```diff
@@ -113,13 +119,50 @@ func (mounter *csiProxyMounter) SMBMount(source, target, fsType string, mountOpt
 		return fmt.Errorf("smb mapping failed with error: %v", err)
 	}
 	klog.V(2).Infof("mount %s on %s successfully", source, normalizedTarget)
+
+	mappingPath := normalizeMappingPath(source)
+	err = incementMappingPathCount(mappingPath, source)
+	if err != nil {
+		klog.Warningf("failed incementMappingPathCount %v", err)
+	}
+
 	return nil
 }
 
 func (mounter *csiProxyMounter) SMBUnmount(target string) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+
 	klog.V(4).Infof("SMBUnmount: local path: %s", target)
-	// TODO: We need to remove the SMB mapping. The change to remove the
-	// directory brings the CSI code in parity with the in-tree.
+
+	if remotePath, err := os.Readlink(target); err != nil {
+		klog.Warningf("SMBUnmount: can't get remote path: %v", err)
+	} else {
+		klog.V(4).Infof("SMBUnmount: remote path: %s", remotePath)
+
+		mappingPath := normalizeMappingPath(remotePath)
+		klog.V(4).Infof("SMBUnmount: mapping path: %s", mappingPath)
```

Comment:
combine logs in L39, L14 as `klog.V(4).Infof("SMBUnmount: remote path: %s, mapping path: %s", remotePath, mappingPath)`

Reply:
done